c7n/resources/cloudfront.py

```diff
@@ -149,6 +150,55 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('distribution-config')
+class DistributionConfig(ValueFilter):
+    """Check for Cloudfront distribution config values
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: logging-enabled
+                resource: distribution
+                filters:
+                  - type: distribution-config
+                    key: Logging.Enabled
+                    value: true
+   """
+
+    schema = type_schema('distribution-config', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ('cloudfront:GetDistributionConfig',)
+    annotation_key = 'c7n:distribution-config'
+    annotate = False
+
+    def process(self, resources, event=None):
+
+        self.augment([r for r in resources if self.annotation_key not in r])
+        return super(DistributionConfig, self).process(resources, event)
+
+    def augment(self, resources):
+
+        client = local_session(self.manager.session_factory).client(
+            'cloudfront', region_name=self.manager.config.region)
+
+        for r in resources:
+            try:
+                r[self.annotation_key] = client.get_distribution_config(Id=r['Id']) \
+                    .get('DistributionConfig')
+            except (client.exceptions.NoSuchResource, client.exceptions.NoSuchDistribution):
+                r[self.annotation_key] = {}
+            except Exception as e:
```

Comment:
looks like this is missing test coverage on error handling here (re patch coverage ci check), this is probably easier via mock or passing in a synthetic resource, as an example of using a mock to check error handling https://github.com/cloud-custodian/cloud-custodian/blob/master/tests/test_cwl.py#L92